### PR TITLE
Initialize location business status merge person

### PR DESCRIPTION
**Pull request by trevlenb2**

### CODE REVIEW

File: src/main/java/com/revealprecision/revealserver/api/v1/controller/EntityTagController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+@RestController
+@RequestMapping("/api/v1/entityTag")
+public class EntityTagController {
+
+  private final EntityTagService entityTagService;
+
+  @Operation(summary = "Create Tag", description = "Create Tag", tags = {"Entity Tags"})
+  @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<EntityTagResponse> createTag(
+      @RequestBody EntityTagRequest entityTagRequest) {
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/api/v1/controller/EntityTagController.java:31]: what happens on empty and or  null values on EntityTagRequest, please add some validation here.

File: src/main/java/com/revealprecision/revealserver/api/v1/dto/factory/EntityTagFactory.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import com.revealprecision.revealserver.persistence.domain.LookupEntityType;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class EntityTagFactory {
+
+  public static EntityTag toEntity(EntityTagRequest entityTagRequest,
+      LookupEntityType lookupEntityType) {
+
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/api/v1/dto/factory/EntityTagFactory.java:15]: if the request does not have validation, you will get null pointers here , so this code assumes validation has happened on the request. please update the request to support this assumption

File: src/main/java/com/revealprecision/revealserver/api/v1/dto/request/EntityTagRequest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.Setter;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class EntityTagRequest {
+
+  private String tag;
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/api/v1/dto/request/EntityTagRequest.java:17]: please add validation

File: src/main/java/com/revealprecision/revealserver/service/BusinessStatusService.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+@RequiredArgsConstructor
+public class BusinessStatusService {
+
+  private final BusinessStatusProperties businessStatusProperties;
+  private final MetaDataJdbcService metaDataJdbcService;
+  private final EntityTagService entityTagService;
+
+  public void setBusinessStatus(Task task, String businessStatus) {
+
+    UUID planIdentifier = task.getAction().getGoal().getPlan().getIdentifier();
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/service/BusinessStatusService.java:28]: has the design been updated to to EAGER fetching of these items?

File: src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+
+  }
+
+  public void createOrUpdateMetadata(UUID entityIdentifier, String entity, String tagName,
+      Object value, Class<?> entityDataType) {
+    String quotes = (entityDataType.equals(String.class) ? "'" : "");
+
+
+// upsert String resolves to below insert statement
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java:53]: are this comments or commented out code?

File: src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+//                'entity_tag'
+//                ,(SELECT identifier from entity_tag where tag = 'business_status' AND lookup_entity_type_identifier =
+//                    (SELECT identifier FROM lookup_entity_type WHERE table_name = 'location'))
+//            ,  'value','NOT_VISITED'
+//        ))
+//				,true
+//			),
+//    WHERE location_metadata.location_identifier = '76a83898-efec-4470-921e-4834d8245c7d'
+
+    String metadataTable = entity + "_metadata";
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java:90]: I'm seeing this for the second time, can be extracted to constant

File: src/main/java/com/revealprecision/revealserver/api/v1/facade/service/TaskFacadeService.java [hunk 7]
```diff
@@ -93,66 +82,29 @@ private User getUser(String createdByUserIdentifier) {
...
 
     UUID identifier = null;
 
     try {
-      Task task = taskService.getTaskByIdentifier(
-          UUID.fromString(updateFacade.getIdentifier()));
+      Task task = taskService.getTaskByIdentifier(UUID.fromString(updateFacade.getIdentifier()));
 
       Optional<LookupTaskStatus> taskStatus = taskService.getAllTaskStatus().stream().filter(
-          lookupTaskStatus -> lookupTaskStatus.getCode()
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/api/v1/facade/service/TaskFacadeService.java:149]: which code formatter are you using?

File: src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MetaDataJdbcService {
+
+  @Autowired
```
Comment (lindagcaba) [inline src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java:22]: maybe constructor injection here...

File: src/main/java/com/revealprecision/revealserver/service/BusinessStatusService.java (hunk 4 above)
Comment (trevlenb2) [reply src/main/java/com/revealprecision/revealserver/service/BusinessStatusService.java:28]: The design allows upwards not downwards so this should be fine..

File: src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java (hunk 5 above)
Comment (trevlenb2) [reply src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java:53]: Wanted to give an example of what the generated SQL will look like....

File: src/main/java/com/revealprecision/revealserver/api/v1/facade/service/TaskFacadeService.java (hunk 7 above)
Comment (trevlenb2) [reply src/main/java/com/revealprecision/revealserver/api/v1/facade/service/TaskFacadeService.java:149]: google one...

File: src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java (hunk 8 above)
Comment (trevlenb2) [reply src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java:22]: agree

File: src/main/java/com/revealprecision/revealserver/api/v1/dto/factory/EntityTagFactory.java (hunk 2 above)
Comment (trevlenb2) [reply src/main/java/com/revealprecision/revealserver/api/v1/dto/factory/EntityTagFactory.java:15]: agree

File: src/main/java/com/revealprecision/revealserver/api/v1/dto/request/EntityTagRequest.java (hunk 3 above)
Comment (trevlenb2) [reply src/main/java/com/revealprecision/revealserver/api/v1/dto/request/EntityTagRequest.java:17]: agree

File: src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java (hunk 6 above)
Comment (trevlenb2) [reply src/main/java/com/revealprecision/revealserver/service/MetaDataJdbcService.java:90]: will consider

**Review by lindagcaba — APPROVED** [review]

**Pull request merged**